### PR TITLE
Edible item fix for discontinued and quest-only

### PR DESCRIPTION
**Pull request by lmsv-mx123**
No longer have to worry on eating or drinking for discontinued rares, quest-only items with drink or eat can get their options swapped.

### CODE REVIEW

File: src/Game/Client.java [hunk 1]
```diff
@@ -757,6 +758,57 @@ public static void checkForUpdate(boolean announceIfUpToDate) {
...
+	 * @param itemName - item name
+	 */
+	public static void redrawMenuHook(Object instance, int n, int index, String itemCommand, String itemName) {
+		
+		if (instance != null) {
+			try {
+				if (!itemCommand.equals("")) {
+					if (!Item.shouldPatch(index)) {
+						Reflection.menuGen.invoke(instance, n, 640, false, itemCommand, "@lre@" + itemName);
+						Reflection.menuGen.invoke(instance, n, 650, false, "Use", "@lre@" + itemName);
```
Comment (OrN) [inline src/Game/Client.java]: Does the text "Use" matches how it does in-game? I noticed earlier your screenshot had "eat".

File: src/Game/Item.java [hunk 2]
```diff
@@ -132,6 +133,42 @@ public static void queryDatabaseAndPatchItem(Connection c, String query) {
 		}
 	}
 	
+	/**
+	 * Patches discontinued edible item commands specified by {@link Settings#COMMAND_PATCH_TYPE}.
+	 * Removes completely the option to eat/drink
+	 */
+	public static void patchItemCommands() {
+		int commandPatchType = Settings.COMMAND_PATCH_TYPE;
+		int[] edible_rare_item_ids = { 246, 387, 422, 677 };
```
Comment (OrN) [inline src/Game/Item.java]: We should leave in a comment what these item IDs are.

File: src/Game/Item.java [hunk 3]
```diff
@@ -132,6 +133,42 @@ public static void queryDatabaseAndPatchItem(Connection c, String query) {
...
+			}
+		}
+	}
+	
+	/**
+	 * Patches quest only edible item commands specified by {@link Settings#COMMANDS_PATCH_TYPE}.
+	 * Swaps around the option to eat/drink
+	 */
+	public static boolean shouldPatch(int index) {
+		int[] edible_quest_item_ids = { 718, 770, 1061, 1086 };
```
Comment (OrN) [inline src/Game/Item.java]: Same as my comment above.

File: src/Game/Client.java (hunk 1 above)
Comment (lmsv-mx123) [reply src/Game/Client.java]: use functionality is exactly the same, just for the part where its swapping it gets temporarily 635, but then in another hook it gets reincorporated as 650

File: src/Game/Item.java (hunk 2 above)
Comment (lmsv-mx123) [reply src/Game/Item.java]: fixed

File: src/Game/Client.java (hunk 1 above)
Comment (OrN) [reply src/Game/Client.java]: I'm talking about the text you set here, is it "Use" or "use"? If it's "use", make sure it reflects that.

Comment (lmsv-mx123) [reply src/Game/Client.java]: its "Use" and was before "Use", that text remains the same

Comment (OrN) [reply src/Game/Client.java]: Ok, and the same is true for the remaining options?

Comment (lmsv-mx123) [reply src/Game/Client.java]: yea they were left as is

**Pull request merged**